src/platform/src/main/java/org/geoserver/platform/resource/FileSystemResourceStore.java

```diff
@@ -429,7 +432,7 @@ public String toString() {
 
         @Override
         public boolean delete() {
-            return file.delete();
+            return file.exists() && Files.delete(file);
         }
```

Comment:
Your feedback is a bit late, we can take it to geotools-devel. The issue with this one is you combined a good fix (that needed to be back ported) with an API change which did not - so I had to split it up.